test/core/pipeline/features/conftest.py

```diff
@@ -34,6 +45,56 @@ class _PipelineResult:
 
 @when("I run the Pipeline", target_fixture="pipeline_result")
 def run_pipeline(
+    pipeline_data: Tuple[Union[AsyncPipeline, Pipeline], List[PipelineRunData]], spying_tracer
+) -> Union[List[Tuple[_PipelineResult, PipelineRunData]], Exception]:
+    if isinstance(pipeline_data[0], AsyncPipeline):
+        return run_async_pipeline(pipeline_data, spying_tracer)
+    else:
+        return run_sync_pipeline(pipeline_data, spying_tracer)
+
+
+def run_async_pipeline(
+    pipeline_data: Tuple[Union[AsyncPipeline], List[PipelineRunData]], spying_tracer
+) -> Union[List[Tuple[_PipelineResult, PipelineRunData]], Exception]:
+    """
+    Attempts to run a pipeline with the given inputs.
+    `pipeline_data` is a tuple that must contain:
+    * A Pipeline instance
+    * The data to run the pipeline with
+
+    If successful returns a tuple of the run outputs and the expected outputs.
+    In case an exceptions is raised returns that.
+    """
+    pipeline, pipeline_run_data = pipeline_data[0], pipeline_data[1]
+
+    results: List[_PipelineResult] = []
+
+    async def run_inner(data, include_outputs_from):
+        return await pipeline.run_async(data=data.inputs, include_outputs_from=include_outputs_from)
+
+    for data in pipeline_run_data:
+        try:
+            async_loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(async_loop)
+            outputs = async_loop.run_until_complete(run_inner(data, data.include_outputs_from))
```

Comment:
Can we replace this with a (preferred) high level method if there is not an explicit need to manage the event loop manually?
Source: https://docs.python.org/3/library/asyncio-eventloop.html
```suggestion
            outputs = asyncio.run(run_inner(data, data.include_outputs_from))

```

Reply:
Yes, makes sense. That test was already structured like this in the original implementation from Madeesh. I thought a bit about it but I see no reason to manage the event loop manually. I'd still run the tasks one by one since some of our fake test components have state that would change the test results when we run them concurrently.